actionpack/lib/action_controller/test_case.rb

```diff
@@ -659,7 +659,9 @@ def process(action, *args)
         @request.assign_parameters(@routes, controller_class_name, action.to_s, parameters)
 
         @request.session.update(session) if session
-        @request.flash.update(flash || {})
+
+        is_request_flash_enabled = @request.respond_to?(:flash)
```

Comment:
Ah right. Maybe we should move the `enable_flash` call to an `ensure` block at `test_banana_redirects_to_https_if_not_https_and_flash_middleware_is_disabled` and maybe this change is not needed anymore